darglint2/config.py

```diff
@@ -78,23 +79,46 @@ def from_string(cls, level: str) -> "LogLevel":
             raise ValueError('Unrecognized log level, "{}"'.format(level))
 
 
-class Configuration(object):
+class Configuration:
+    """
+    A dataclass representing a configuration.
+
+    Attributes:
+        ignore: A list of error codes to ignore.
+        message_template: the template with which to format the errors.
+        style: The style of docstring.
+        strictness: The minimum strictness to allow.
+        ignore_regex: A regular expression which enables ignoring
+            functions/methods by name.
+        ignore_raise: A list of exceptions that don't need to be
+            documented.
+        ignore_properties: Bool indicating whether to ignore properties functions
+            or not.
+        enable: A list of of error codes that are disabled by default.
+        indentation: The number of spaces to count as an indent.
+        assert_style: The assert style to use (e.g. log on failed
+            assertions, or raise exception on failed assertions.)
+        log_level: Minimum level to log. All other log entries will be filtered out.
+
+    """
+
     def __init__(
         self,
-        ignore: List[str],
-        message_template: Optional[str],
-        style: DocstringStyle,
-        strictness: Strictness,
+        *,
+        ignore: List[str] = [],
```

Comment:
Should not use mutable default value. Better to use `Optional[List[str]] = None` and `self.ignore = ignore or []`.

Same for `ignore_raise` and `enable` below.